pkgs/top-level/release-outpaths-parallel.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -x
+CHECK_META=$1
+
+nix-instantiate --eval --strict --json pkgs/top-level/release-attrpaths-superset.nix -A names > release-attrpaths-superset.out
+
+NUM_CHUNKS=$((4*$NIX_BUILD_CORES))
+
+parallel -j $NIX_BUILD_CORES \
```

Comment:
Did you try nix-eval-jobs, it does parallel but doesn't help with memory usage AFAIK

Reply:
It helps with memory usage as well because it restarts workers if they take too much memory. The chunking at the moment might be less ideal because it doesn't chunk the list into X parts and gives this to threads but instead starts to iterate the list from the beginning and give each thread an attribute in lexical key order. I think it would be faster if attributes closer to each other would be evaluated by the same thread to avoid evaluating the same thing twice (i.e. x86_64-linux packages evaluated on thread1 and aarch64-linux packages on thread2).

Reply:
However from my tests I would say that having more threads than CPUs will actually slow you down because you end up evaluating the same thing more. I usually have less threads than cores for large package sets like nixpkgs to avoid duplicating work and rather trait it for memory.